### ISSUE TITLE
Add copious comments for new programmers in the source code

### ISSUE THREAD

**Issue by ntoll** [tjguk/networkzero#4]
As a new programmer who has just found their feet with Python and network zero I'd like to be able to look at the source and understand (at a high level) how it works. I can't stress how important it is that we foster the attitude of autonomous and autodidactic research. ;-)

I'm happy to help with this effort.

**Comment by tjguk** [comment]
Very much up for this, @ntoll. I'd wait a little because I'm still moving some of the pieces around (especially after several people last night pointed out that getaddrinfo on Linux works quite differently from under Windows!). But I'm definitely +1 on the idea and I'll try to add commenting in as I go.

**Comment by ntoll** [comment]
Exactly! I imagine the next few weeks will involved some refactoring to get towards a 0.9 version - at which point documentation and code commenting would be a useful thing to do to get it over the line for 1.0. :-)
